libs/output-mapping/src/Writer/TableWriter.php

```diff
@@ -205,7 +206,10 @@ private function createLoadTableTask(
         // - columns in config + headless CSV (SAPI always expect to have a header in CSV)
         // - sliced files
         if ($createTypedTables && !$destinationTableExists && ($hasColumns && !empty($config['column_metadata']))) {
-            $tableDefinitionFactory = new TableDefinitionFactory();
+            $tableDefinitionFactory = new TableDefinitionFactory(
+                array_key_exists('metadata', $config) ? $config['metadata'] : [],
```

Comment:
is this any different to $config['metadata'] ?? []  ?

Reply:
I doubt it 😄